src/test/TestUtils/TestProjectFixture.cs

```diff
@@ -277,16 +277,23 @@ private void ValidateRequiredDirectories(RepoDirectoriesProvider repoDirectories
             return this;
         }
 
-        public TestProjectFixture RestoreProject(string[] fallbackSources)
+        public TestProjectFixture RestoreProject(string[] fallbackSources, string extraMSBuildProperties = null)
         {
             var restoreArgs = new List<string>();
             foreach (var fallbackSource in fallbackSources)
             {
-                restoreArgs.Add("-f");
+                restoreArgs.Add("--source");
                 restoreArgs.Add(fallbackSource);
             }
             restoreArgs.Add("--disable-parallel");
 
+            restoreArgs.Add($"/p:MNAVersion={_repoDirectoriesProvider.MicrosoftNETCoreAppVersion}");
```

Comment:
Actually, I'm going to keep it as-is (unless you still disagree). I'll try to explain the rationale more clearly:

1) These projects are batch-restored on the command line, and it would be nice to pass the same parameters to them all
2) The two "real" test projects (HostActivationTests and DependencyModel) do not want this property to be set, while the rest of the test assets do
3) If we pass it in as a global property, we can't "undefine" it in the two projects that we don't want it in.